syntax = "proto3";
package proto;

import "FileInfo.proto";
import "OpenFile.proto";
import "defs.proto";
import "Histogram.proto";
import "ImageData.proto";
import "SpectralProfile.proto";
import "SpatialProfile.proto";
import "Region.proto";



service FileService{
    rpc CheckStatus(Empty) returns  (StatusResponse) {}
    rpc OpenFile(OpenFileRequest) returns (OpenFileACK) {}
    rpc CloseFile(FileCloseRequest) returns (StatusResponse) {}
    rpc GetFileInfo (FileInfoRequest) returns (FileInfoResponse) {}
    rpc CreateRegion (SetRegion) returns (SetRegionAck) {}
    rpc GetImageDataStream (ImageDataRequest) returns (stream ImageDataResponse) {}
    rpc GetSpatialProfile (SetSpatialReq) returns (SpatialProfileData) {}
    rpc GetSpectralProfile (SpectralProfileRequest) returns (SpectralProfileResponse) {} 
    rpc GetHistogram (SetHistogramReq) returns (HistogramResponse) {}
  }